### PR TITLE
:new:  bot monitoring via jolokia and openshift service/route

### PR DESCRIPTION
**Pull request by maoo**
Still missing `livenessProbe` and `readinessProbe` configuration, but at least now we have a URL to monitor the bot:

```
curl http://bot-github-chatops-dev-jolokia-ssf-dev.b9ad.pro-us-east-1.openshiftapps.com/jolokia/exec/org.symphonyoss.client:type=ClientCheckMBean/isUp | grep '"value":true'
echio $?
```

I'll send another PR as soon as I figure out openshift probe configuration.

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -25,11 +25,12 @@
                      :url                     "http://www.apache.org/licenses/LICENSE-2.0"}
   :min-lein-version "2.8.1"
   :repositories     [["sonatype-snapshots" {:url "https://oss.sonatype.org/content/groups/public" :snapshots true}]
-                     ["jitpack"            {:url "https://jitpack.io"}]]
+                    ["jitpack" "https://jitpack.io"]]
```
Comment (pmonks) [inline project.clj]: Please preserve horizontal alignment in statically initialised data structures (in this case the `:repositories` vector).

File: project.clj [hunk 2]
```diff
@@ -25,11 +25,12 @@
...
   :plugins          [
                       [org.noisesmith/git-info-edn "0.2.1"]
                     ]
   :dependencies     [
+                      [com.github.maoo.symphony-java-client/symphony-client "mbeans-rc-1" :exclusions [org.slf4j/slf4j-log4j12]]
```
Comment (pmonks) [inline project.clj]: When do you expect SJC to merge your changes upstream?  I have a strong preference for waiting for that to happen, rather than having the bot add dependencies to a fork.

File: resources/config.edn [hunk 3]
```diff
@@ -25,8 +25,8 @@
   ; For details, see https://github.com/rhuss/jolokia/blob/master/agent/jvm/src/main/resources/default-jolokia-agent.properties
   ; Note: All keys and values MUST be strings - this is a Jolokia requirement
   :jolokia-config {
-    "host"  #or [#env JOLOKIA_HOST  "localhost"]
-    "port"  #or [#env JOLOKIA_PORT  "8778"]
+    "host"  #or [#env JOLOKIA_HOST "*"]
```
Comment (pmonks) [inline resources/config.edn]: This should be configured in the OpenShift environment, not in the default `config.edn` file, since opening up Jolokia in this way is potentially a massive security hole (so that should not be the default).

File: resources/jolokia-access.xml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<restrict>
+    <remote>
+        <host>localhost</host>
+        <host>10.0.0.0/8</host>
+    </remote>
+
+     <http>
```
Comment (pmonks) [inline resources/jolokia-access.xml]: Alignment is off.

File: resources/jolokia-access.xml [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (pmonks) [inline resources/jolokia-access.xml]: XML files should use indentation of 2 spaces.

File: resources/logback.xml [hunk 6]
```diff
@@ -15,8 +15,8 @@
 
   <logger name="org.apache"      level="ERROR" />
   <logger name="org.symphonyoss" level="ERROR" />
-  <logger name="org.jolokia"     level="ERROR" />
-
+  <logger name="org.jolokia"     level="INFO" />
+  <logger name="org.symphonyoss.client.impl.SymphonyBasicClient" level="INFO" />
```
Comment (pmonks) [inline resources/logback.xml]: Is this a hangover from development / debugging?  This should not be the default logging setting.

File: resources/config.edn (hunk 3 above)
Comment (pmonks) [reply resources/config.edn]: Also, need to preserve horizontal alignment here too.

File: .travis.yml [hunk 7]
```diff
@@ -20,7 +20,7 @@ deploy:
     skip_cleanup: true
     script: lein do git-info-edn, uberjar && mkdir -p target/oc && cp target/*-standalone.jar target/oc/ && ./deploy-to-openshift.sh $TRAVIS_BRANCH $TRAVIS_PULL_REQUEST
     on:
-      all_branches: true
+      branch: dev
```
Comment (pmonks) [inline .travis.yml]: Doesn't this change prevent merges to master from being deployed to OpenShift?

File: project.clj (hunk 2 above)
Comment (maoo) [reply project.clj]: They are already merged to dev branch and deployed to `symphony-client-1.1.3-SNAPSHOT` , see https://travis-ci.org/symphonyoss/symphony-java-client . I will test with the SNAPSHOT version and keep you posted.

File: resources/logback.xml (hunk 6 above)
Comment (maoo) [reply resources/logback.xml]: Jolokia log level is, but the SJC is IMO useful, as it logs if/when the JMX MBean is exposed; I'll take it out if you don't like it.

File: .travis.yml (hunk 7 above)
Comment (maoo) [reply .travis.yml]: No, it prevents all branches to deploy on OpenShift (ssf-dev project), for example the `mbeans` branch. The `master`  branch deployment configuration is defined in the block below - see https://github.com/symphonyoss/bot-github-chatops/blob/master/.travis.yml#L26

Without this change, all Travis PR builds would fail, since travis tries to run the `./deploy-to-openshift.sh` script, which fails because it only knows how to deploy `master` and `dev` branches.

Comment (pmonks) [reply .travis.yml]: Where is deploy-to-openshift.sh called in the next `provider` block?  I don't see it...

Also, shouldn't deploy-to-openshift.sh be more defensive, and silently exit if it doesn't recognise the branch name?  That would be a more robust approach.

File: resources/logback.xml (hunk 6 above)
Comment (pmonks) [reply resources/logback.xml]: I've found SJC to be too noisy to be left at `INFO` level, though I realise you're only setting it on a single class.

Regardless, best practice is to set logging to `ERROR` only by default - it can always be overridden in any particular environment when needed (including via JMX!).

**Comment by maoo** [comment]
@pmonks - addressed all issues, except the upstream dependency; let's keep this PR open until we can define a dependency to an SJC SNAPSHOT.

File: project.clj [hunk 8]
```diff
@@ -25,7 +25,7 @@
                      :url                     "http://www.apache.org/licenses/LICENSE-2.0"}
   :min-lein-version "2.8.1"
   :repositories     [["sonatype-snapshots" {:url "https://oss.sonatype.org/content/groups/public" :snapshots true}]
-                     ["jitpack"            {:url "https://jitpack.io"}]]
+                     ["jitpack" "https://jitpack.io"]]
```
Comment (pmonks) [inline project.clj]: Still need to revert this change.

File: .travis.yml (hunk 7 above)
Comment (maoo) [reply .travis.yml]: Added condition on travis build - https://github.com/symphonyoss/bot-github-chatops/commit/72d92562f9319fa6165990eb1b8f928169bf4a5c

Re. deploy-to-openshift.sh, I'm still checking how to make https://github.com/symphonyoss/contrib-toolbox/blob/master/scripts/oc-deploy.sh more defensive

**Comment by ssf-admin** [comment]
@pmonks - I think the PR is ready, fixed readiness and liveness probe. Do we want to wait for SJC 1.1.3 release, or you're ok using the 1.1.3-SNAPSHOT version, at least to merge on `dev` branch?
